### PR TITLE
Replacing `check_run` validation by `status check`

### PR DESCRIPTION
**Pull request by Oluane**
## Description

This PR aims at fixing some issues we had with the previous implementation of `check_run` validation of our e2e tests in CI. The workaround was to replace `check_run` usage by the `status check`, which is the other kind of PR validation provided by Github API. We create this `status check` right after the e2e tests suite execution, and set its state with tests results, along with the `target_url`. It allows to display a link leading to the job (where we can re-run tests if needed) in the checks section in PR UI. 

The previous issues happening with our first implementation doesn't seem to occur with the `status checks`. 

## Type of change

- [ ] **Chore** (non-breaking change which refactors / improves the existing code base).
- [x] **Bug fix** (non-breaking change which fixes an issue).
- [ ] **New feature** (non-breaking change which adds functionality).
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected).

## How Has This Been Tested?

Manually.

## Checklist:

- [x] My code follows the style [**contributing guidelines**][contributing_file] of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.

[contributing_file]: https://github.com/fewlinesco/connect-account/blob/master/README.adoc

### CODE REVIEW

**Review by Fuzznimp — COMMENTED** [review]
Good for me 👍.

**Review by Fenntasy — COMMENTED** [review]
Ok if I understand correctly, the way it works with this code is:
- nothing happens when we run the tests
- we create a status failed or success when we run the e2e tests

Am I right? Couldn't we create a status as pending when running?

**Comment by Fenntasy** [comment]
> I didn't do it that way in first place because I noticed that when you mark a check/status as required, it's automatically displayed as pending in UI check section, even if it is linked to nothing at the moment. 🙂

oh, well ok then 🙇

**Review by Fenntasy — DISMISSED** [review]

File: README.adoc [hunk 1]
```diff
@@ -474,18 +474,16 @@ We are using https://github.com/getgauge/taiko[Taiko] to test our workflows.
...
-To do so, we first create manually a https://docs.github.com/en/rest/reference/checks#get-a-check-run[check-run] (`init-e2e-test-check` job inside `lint-and-test.yml`). This custom `check-run` is just initialized at this point, and its `status` is pending.
-Once we receive the status of the deployment from our third party actor, we run our end to end test suite (`e2e-tests-heroku` job inside `e2e-test-heroku.yml`) if the review app deployment is a success, or we skip it in the case of a failure during the process.
-The last job (`e2e-tests-check-run-status` inside `e2e-test-heroku.yml`) is used to manage the status of the custom `check-run` we created previously. 
-If the deployment failed, we set its status to failure too. When the end to end test suite can be runned, we wait for it to complete before setting the custom `check-run` status according to the tests results. 
+Once we receive the status of the deployment from our third party actor, we run our end to end test suite (`e2e-tests-heroku.yml` job inside `e2e-test-heroku.yml`) if the review app deployment is a success, or we skip it in the case of a failure during the process.
```
Comment (Fenntasy) [inline README.adoc:479]:
```suggestion
Once we receive the status of the deployment from our third party actor, we run our end to end test suite (`e2e-tests` job inside `e2e-test.yml`) if the review app deployment is a success, or we skip it in the case of a failure during the process.
```

File: README.adoc [hunk 2]
```diff
@@ -474,18 +474,16 @@ We are using https://github.com/getgauge/taiko[Taiko] to test our workflows.
...
+In the last job (`create-e2e-status-check` inside `e2e-test-heroku.yml`), we create manually a https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-status-checks[status check].
```
Comment (Fenntasy) [inline README.adoc:480]:
```suggestion
In the last job (`create-e2e-status-check` inside `e2e-test.yml`), we create manually a https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-status-checks[status check].
```

**Review by Fenntasy — APPROVED** [review]

**Pull request merged**